cpp/src/connect/weak_cc.cuh

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
```

Comment:
It seems like cuDF folks are removing #pragma once in favor of the good old-style include guards.

https://github.com/rapidsai/cudf/issues/578

We may better change this all at once in another issue but just FYI.

Reply:
I am strongly against old-style include guards (which I find anything but _good_), because:
(1.) Name pollution (pre-processor space name pollution, but still...);
(2.) There's no way to prevent misspelling the pre-processor name. I've seen things like:

#ifndef SOME_NAME
#define SOME_NAME_but_slightly_different
...
#endif

Reply:
I think we can live with ` #pragma once`. It comes with advantages and it is not causing any problems at this point.

Reply:
OK, I have no preference on one over the other. I am sometimes (admittedly, overly) obsessed with consistency, and I personally lean towards following cudf unless there is a strong reason to do not (in this case, we may better bring the issue back to cudf). But I guess this will hardly become a problem in practice, and let's wait till we really see an issue with mixing the two styles.